components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java

```diff
@@ -272,5 +334,145 @@ protected User getUser(Map eventProperties, UserStoreManager userStoreManager){
         return user;
     }
 
+    private void preSetUserClaimsOnEmailUpdate(Map<String, String> claims, UserStoreManager userStoreManager, User user)
+            throws IdentityEventException {
+
+        Utils.clearEmailVerifyTemporaryClaim();
+
+        if (MapUtils.isEmpty(claims)) {
+            // Not required to handle in this handler.
+            return;
+        }
+
+        String verifyEmailClaimValue = claims.get(IdentityRecoveryConstants.VERIFY_EMAIL_CLIAM);
+        if (StringUtils.isNotBlank(verifyEmailClaimValue)) {
+
+            Claim verifyEmailClaim = new Claim();
+            verifyEmailClaim.setClaimUri(IdentityRecoveryConstants.VERIFY_EMAIL_CLIAM);
+            verifyEmailClaim.setValue(verifyEmailClaimValue);
+            Utils.setEmailVerifyTemporaryClaim(verifyEmailClaim);
+            // Remove claim to prevent persisting this temporary claim.
+            claims.remove(IdentityRecoveryConstants.VERIFY_EMAIL_CLIAM);
+
+            String emailAddress = claims.get(IdentityRecoveryConstants.EMAIL_ADDRESS_CLAIM);
 
+            if (isVerificationPendingEmailClaimConfigAvailable(user.getTenantDomain()) &&
+                    StringUtils.isNotBlank(emailAddress)) {
+
+                String existingEmail = null;
+                String username = IdentityUtil.addDomainToName(user.getUserName(), user.getUserStoreDomain());
+                try {
+                    existingEmail = userStoreManager.getUserClaimValue(username,
+                            IdentityRecoveryConstants.EMAIL_ADDRESS_CLAIM, null);
+                } catch (UserStoreException e) {
+                    log.error("Error occurred while retrieving existing email address for user: " + username, e);
+                }
+
+                if (emailAddress.equals(existingEmail)) {
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format("The email address to be updated is same as the existing email " +
```

Comment:
`The updated email address is the same as the existing email address for user: %s . Hence an email verification will not be triggered`